taipy/core/data/_file_datanode_mixin.py

```diff
@@ -135,7 +135,16 @@ def _upload(self, path: str, upload_checker: Optional[Callable[[str, Any], bool]
             return reason_collection
 
         if upload_checker is not None:
-            if not upload_checker(upload_path.name, upload_data):
+            try:
+                can_upload = upload_checker(upload_path.name, upload_data)
+            except Exception as err:
```

Comment:
Should the exception be added to the reason?
Especially useful on user function

Reply:
We could add an optional exception attached to a reason. It is useful information for developers. It is already in the logs, but making it available in taipy so they can use it in the code makes sense to me.

However, by default, I would not display it in the data node viewer to end-users. 

What do you think?

Reply:
There is only 1 way of returning a Reason, so I think it's the default.

I'm not sure if we can give the user multiple choices